src/main/java/org/broadinstitute/hellbender/tools/ApplyBQSR.java

```diff
@@ -51,6 +53,13 @@ public void onTraversalStart() {
         final SAMFileHeader outputHeader = ReadUtils.cloneSAMFileHeader(getHeaderForReads());
         outputWriter = new SAMFileGATKReadWriter(new SAMFileWriterFactory().makeWriter(outputHeader, true, OUTPUT, referenceArguments.getReferenceFile()));
         transform = new BQSRReadTransformer(outputHeader, BQSR_RECAL_FILE, bqsrArgs);
+        assertOnlyIlluminaReadGroups(getHeaderForReads());
+    }
+
+    private void assertOnlyIlluminaReadGroups(final SAMFileHeader readsHeader) {
+        if (readsHeader.getReadGroups().stream().anyMatch(rg -> NGSPlatform.fromReadGroupPL(rg.getPlatform()) !=  NGSPlatform.ILLUMINA)){
+            throw new UserException.BadInput("BQSR only supports ILLUMINA.");
```

Comment:
Rather than throwing an exception, I would log a warning to the user saying that only Illumina-based sequencing is well tested.  But we have shown that recalibration works on many different platforms, so there's no reason not to allow a user to attempt it.